trieste/models/model_interfaces.py

```diff
@@ -589,10 +607,19 @@ def _assert_data_is_compatible(new_data: Dataset, existing_data: Dataset) -> Non
 
 def randomize_model_hyperparameters(model: gpflow.models.GPModel) -> None:
     """
-    Sets hyperparameters to random samples from their prior distributions.
+    Sets hyperparameters to random samples from their constrained domains or (if not constraints
+    are available) their prior distributions.
 
     :param model: Any GPModel from gpflow.
     """
-    for parameter in model.trainable_parameters:
-        if parameter.prior is not None:
-            parameter.assign(parameter.prior.sample())
+    for param in model.trainable_parameters:
+        if isinstance(param.bijector, tfp.bijectors.Sigmoid):
```

Comment:
do we have other possible transformations? or is Sigmoid the only case to consider?